frontend/src/metabase/query_builder/components/view/NativeQueryPreviewSidebar/NativeQueryPreviewSidebar.tsx

```diff
@@ -49,18 +53,63 @@ export const NativeQueryPreviewSidebar = (): JSX.Element => {
   const getErrorMessage = (error: unknown) =>
     typeof error === "string" ? error : undefined;
 
+  const borderStyle = `1px solid ${color("border")}`;
+  const aceMode = getEngineNativeType(engineType);
+
   return (
-    <NativeQueryPreview
-      title={TITLE[engineType]}
-      query={query}
-      error={getErrorMessage(error)}
-      isLoading={isLoading}
+    <Flex
+      className={cx(SB.container)}
+      role="complementary"
+      direction="column"
+      style={{ borderLeft: borderStyle }}
     >
-      {query && (
-        <Button variant="subtle" onClick={handleConvertClick}>
-          {BUTTON_TITLE[engineType]}
-        </Button>
-      )}
-    </NativeQueryPreview>
+      <Box
+        component="header"
+        c={color("text-dark")}
+        fz={rem(20)}
+        lh={rem(24)}
+        fw="bold"
+        ta="start"
+        p="1.5rem"
+      >
+        {TITLE[engineType]}
+      </Box>
+      <Box
+        style={{ flex: 1, borderTop: borderStyle, borderBottom: borderStyle }}
+      >
+        {isLoading && <DelayedLoadingSpinner delay={1000} />}
+        {error && (
+          <Flex align="center" justify="center" h="100%" direction="column">
+            <Icon name="warning" size="2rem" color={color("error")} />
+            {t`Error generating the query.`}
+            <Box mt="sm">{getErrorMessage(error)}</Box>
+          </Flex>
+        )}
+        {!error && query && (
+          <NativeQueryEditorRoot style={{ height: "100%", flex: 1 }}>
```

Comment:
I am not particularly fond of this approach, but all ace style overrides are defined in `NativeQueryEditorRoot` and we want styles to be the same here.

I'll follow up with a PR that extracts those styles somewhere that makes access to it easier for any component.

Reply:
`aceEditorStyle` can be moved here https://github.com/metabase/metabase/blob/db2624f86ac0f334349f19f7fc3c8b850256c654/frontend/src/metabase/styled-components/containers/GlobalStyles/GlobalStyles.tsx#L35

Reply:
So you won't need to use this component

Reply:
I think you can move them to `GlobalStyles` for now because there are other ace editor styles there

Reply:
and them we can refactor this part in `GlobalStyles`